sphinx/ext/autodoc/__init__.py

```diff
@@ -1508,7 +1508,6 @@ def add_directive_header(self, sig: str) -> None:
         sourcename = self.get_sourcename()
         if inspect.isabstractmethod(self.object):
             self.add_line('   :abstractmethod:', sourcename)
-        self.add_line('   :property:', sourcename)
```

Comment:
I don’t understand, what do you mean?

Reply:
It is better to generate `:type:` option if target property has an annotation for return value:
```
@property
def prop(self) -> int:
    """blah blah blah"""
```
This will converted into 
```
.. py:property:: prop
   :type: int

   blah blah blah
```

Reply:
OK, sure, thank you!

Reply:
Hmm, actually properties can have a different return type and “settable” type, e.g.:

```py
@property
def path(self) -> Path:
    return self._path

@path.setter
def path(self, value: typing.PathLike):
    self._path = Path(value)
```

I don’t know how to best represent this. For now just the return type might be enough.